.github/workflows/ci.yaml

```
@@ -24,19 +24,25 @@ jobs:
       - uses: actions/checkout@v5
         with:
           fetch-depth: 0
+      - name: Calculate reference for git diff
+        id: ref
+        if: github.event_name == 'pull_request'
+        run: |
+          set -xueo pipefail
+          echo "result=$(git merge-base origin/${{ github.event.pull_request.base.ref }} HEAD)" >> "$GITHUB_OUTPUT"
```

I've just merged `main`, and the idea of moving this logic to `get-changed.py` is no longer immediately viable, as there's now another location where we need to use this ref -- a direct call to `ls.py`.

We don't call `get-changed.py` in that case because we don't want the functionality it layers on top of `ls.py`, namely detecting changes to 'global' files (we don't care about them, we only care about which packages would be published).

If you're not happy with it in the current state, would you prefer this? It's a bit more verbose but perhaps makes the non `pull_request` event behaviour more explicit.
```suggestion
      - name: Calculate reference for git diff
        id: ref
        run: |
          set -xeo pipefail
          if [ ${{ github.event_name }} = 'pull_request' ]; then
            REF=$(git merge-base origin/${{ github.event.pull_request.base.ref }} HEAD)
          else
            REF=
          fi
          echo "result=$REF" >> "$GITHUB_OUTPUT"
```

---

@dimaqq tagging you in case you don't see this message since it's not a direct reply.

---

No it's ok, I know you didn't suggest these approaches, I just wondered if you'd prefer them. Thanks for the review.